### PR TITLE
Fix #8441: Consider adding an overload for GetRevisionsAsync that takes key and label

### LINKED ISSUE #8441: Consider adding an overload for GetRevisionsAsync that takes key and label.
Alternatively, it could take a ConfigurationSetting.  

See: https://github.com/Azure/azure-sdk-for-net/pull/8330/files#r337800482

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs [hunk 1]
```diff
@@ -561,6 +561,45 @@ await foreach (ConfigurationSetting value in service.GetRevisionsAsync(selector,
...
+                AsyncPageable<ConfigurationSetting> revisions = service.GetRevisionsAsync(testSettingUpdate.Key, testSettingUpdate.Label, CancellationToken.None);
+
+                int resultsReturned = 0;
+                await foreach (ConfigurationSetting value in revisions)
+                {
+                    Assert.True(ConfigurationSettingEqualityComparer.Instance.Equals(value, testSettingUpdate));
+                    resultsReturned++;
+                }
+
+                Assert.AreEqual(1, resultsReturned);
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs]: Do we have any tests that get multiple revisions?  It doesn't have to be for this overload, but if not, we should add one.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 2]
```diff
@@ -626,6 +626,30 @@ public virtual Pageable<ConfigurationSetting> GetConfigurationSettings(SettingSe
             return PageResponseEnumerator.CreateEnumerable(nextLink => GetConfigurationSettingsPage(selector, nextLink, cancellationToken));
         }
 
+        /// <summary>
+        /// Retrieves the different revisions of specific <see cref="ConfigurationSetting"/> with specified <paramref name="key"/> and <paramref name="label"/>.
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: "of a specific"
"with the specified"

I realize this was copied from elsewhere - would you mind updating those instances as well?

**Review by annelo-msft — APPROVED** [review]

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs (hunk 1 above)
Comment (AlexanderSher) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs]: `GetRevisions` test checks for several revisions.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 2 above)
Comment (AlexanderSher) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: Fixed

**Pull request merged**